src/Http/Headers/test/MediaTypeHeaderValueTest.cs

```diff
@@ -683,6 +683,85 @@ public void TryParseStrictList_WithSomeInvalidValues_ReturnsFalse()
             Assert.False(MediaTypeHeaderValue.TryParseStrictList(inputs, out var results));
         }
 
+        [Theory]
+        [InlineData("*/*;", "*/*")]
+        [InlineData("text/*", "text/*")]
+        [InlineData("text/*", "text/plain")]
+        [InlineData("*/*;", "text/plain")]
+        [InlineData("text/plain", "text/plain")]
+        [InlineData("text/plain;", "text/plain")]
+        [InlineData("text/plain;", "TEXT/PLAIN")]
+        [InlineData("application/json;v=2", "application/json;param=4")]
+        public void MatchesMediaType_PositiveCases(string mediaType1, string mediaType2)
+        {
+            // Arrange
+            var parsedMediaType1 = MediaTypeHeaderValue.Parse(mediaType1);
+
+            // Act
+            var matches = parsedMediaType1.MatchesMediaType(mediaType2);
+
+            // Assert
+            Assert.True(matches);
+        }
+
+        [Theory]
+        [InlineData("application/html", "text/*")]
+        [InlineData("application/json", "application/html")]
+        [InlineData("text/plain;", "*/*")]
+        public void MatchesMediaType_NegativeCases(string mediaType1, string mediaType2)
+        {
+            // Arrange
+            var parsedMediaType1 = MediaTypeHeaderValue.Parse(mediaType1);
+
+            // Act
+            var matches = parsedMediaType1.MatchesMediaType(mediaType2);
+
+            // Assert
+            Assert.False(matches);
+        }
+
+        [Theory]
+        [InlineData("application/entity+json", "application/entity+json")]
+        [InlineData("application/*+json", "application/entity+json")]
+        [InlineData("application/*+json", "application/*+json")]
+        [InlineData("application/problem+json", "application/json")]
+        [InlineData("application/vnd.restful+json", "application/json")]
```

Comment:
These seem backwards. json is a superset of foo+json, not a subset of.

Reply:
test case fixes.